llm/gguf.go

```diff
@@ -618,22 +618,8 @@ func (llm *gguf) Encode(ws io.WriteSeeker, kv KV, tensors []Tensor) error {
 		}
 	}
 
-	offset, err := ws.Seek(0, io.SeekCurrent)
```

Comment:
making this change here because using encode produces a different result compared to writing out the gguf by hand. Encode writes out the padding even if there aren't any KVs or Tensors which is unnecessary. Instead, write the padding before writing out the tensor which also simplifies the code